httpx/utils.py

```diff
@@ -20,6 +20,21 @@ def normalize_header_value(value: typing.AnyStr, encoding: str = None) -> bytes:
     return value.encode(encoding or "ascii")
 
 
+def str_query_param(value: typing.Union[str, int, float, bool, type(None)]) -> str:
```

Comment:
are there any obstacles to using PrimitiveData? for consistency

Reply:
Yes and no. We'd need some tweaking to avoid a cyclical dependency.